dev/requirements.txt

```diff
@@ -53,7 +53,7 @@ py
 # Spark Connect (required)
 grpcio>=1.48,<1.57
 grpcio-status>=1.48,<1.57
-protobuf==3.20.3
+protobuf==4.25.1
```

Comment:
Hmmm..  I remember we set this protobuf to 3.X because of some compat problem ... do you remember @grundprinzip ?

Reply:
From https://github.com/apache/spark/pull/38693

Reply:
If then, we may need to drop `Python 3.11` support officially from `Spark Connect` module.

Reply:
We had better match the server and client via 4.25.1 from Apache Spark 4.0.0. Otherwise, it's difficult to support Python 3.11 and Python 3.12 and more.